cmd/snap-update-ns/change.go

```diff
@@ -57,28 +58,180 @@ func (c Change) String() string {
 // changePerform is Change.Perform that can be mocked for testing.
 var changePerform func(*Change) ([]*Change, error)
 
-// changePerformImpl is the real implementation of Change.Perform
-func changePerformImpl(c *Change) ([]*Change, error) {
-	if c.Action == Mount {
-		const (
-			mode = 0755
-			uid  = 0
-			gid  = 0
-		)
-		// Create target mount directory if needed.
-		if err := ensureMountPoint(c.Entry.Dir, mode, uid, gid); err != nil {
-			return nil, err
+func (c *Change) createInodes(path, kind string, pokeHoles bool) ([]*Change, error) {
+	var err error
+	var changes []*Change
+
+	// In case we need to create something, some constants.
+	const (
+		mode = 0755
+		uid  = 0
+		gid  = 0
+	)
+
+	// If the element doesn't exist we can attempt to create it.  We will
+	// create the parent directory and then the final element relative to it.
+	// The traversed space may be writable so we just try to create things
+	// first.
+
+	// TODO: refactor this, if possible, with inspection and pre-emptive
+	// creation after the current release ships. This should be possible but
+	// will affect tests heavily (churn, not safe before release).
+retry:
+	switch kind {
```

Comment:
```
for pokeHoles {
    // ... all content below before the return ...
    break
}
return changes, err
```

Reply:
Wow, I didn't know this is doable from bool :D